### PR TITLE
Collection initializer with an expression.

### PR DESCRIPTION
**Pull request by LucianoPAlmeida**
An convenience init to create a collection of a given size initialized with the values of an expression :))

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Tests/SwiftStdlibTests/RangeReplaceableCollectionTests.swift [hunk 1]
```diff
@@ -11,6 +11,13 @@ import XCTest
 
 class RangeReplaceableCollectionTests: XCTestCase {
 
+    func testInitExpressionOfSize() {
```
Comment (BennX) [inline Tests/SwiftStdlibTests/RangeReplaceableCollectionTests.swift]: I'm missing the test for condition the size negative or 0 within the test. Can we add that?

File: Tests/SwiftStdlibTests/RangeReplaceableCollectionTests.swift [hunk 2]
```diff
@@ -11,6 +11,13 @@ import XCTest
...
+        var array = [1, 2, 3]
+        let newArray = [Int](expression: array.removeLast(), size: array.count)
```
[the remark this answers is not in the document]
Comment (LucianoPAlmeida) [reply Tests/SwiftStdlibTests/RangeReplaceableCollectionTests.swift]: Crash, but I believe it's something to user handle. We can't control what is inside the expression :))

File: Sources/Extensions/SwiftStdlib/RangeReplaceableCollectionExtensions.swift [hunk 3]
```diff
@@ -6,6 +6,29 @@
...
+    /// of a call of the given expression.
+    ///
+    ///     let values = Array(expression: "Value", count: 3)
+    ///     print(values)
+    ///     // Prints "["Value", "Value", "Value"]"
+    ///
+    /// - Parameters:
+    ///   - expression: The expression to execute for each position of the collection.
+    ///   - size: The size of the collection.
+    public init(expression: @autoclosure () -> Element, size: Int) {
```
[the remark this answers is not in the document]
Comment (LucianoPAlmeida) [reply Sources/Extensions/SwiftStdlib/RangeReplaceableCollectionExtensions.swift]: @guykogus  Such a constructor already exists on the standard lib called `init(repeating:count)` and in fact it a use case of that was the reason for me to have to create this one :))
Here's my use case bellow:
screen shot 2018-08-08 at 7 45 19 am

[the remark this answers is not in the document]
Comment (guykogus) [reply Sources/Extensions/SwiftStdlib/RangeReplaceableCollectionExtensions.swift]: Ugh, how did I miss that?! 🤦‍♂️
